cnf-certification-test/networking/suite.go

```diff
@@ -179,3 +184,51 @@ func testNetworkConnectivity(env *provider.TestEnvironment, aIPVersion netcommon
 		ginkgo.Fail(fmt.Sprintf("%d nets failed the %s network %s ping test.", n, aType, aIPVersion))
 	}
 }
+
+func testOCPReservedPortsUsage(env *provider.TestEnvironment) {
+	// List of all ports reserved by Openshift
+	OCPReservedPorts := map[int32]bool{22623: true, 22624: true}
+
+	var failedContainers int
+	var rogueContainers []string
+	for _, cut := range env.Containers {
+		// First check if any of the containers under test has declared a port reserved by OCP
+		for _, port := range cut.Data.Ports {
+			if OCPReservedPorts[port.ContainerPort] {
+				tnf.ClaimFilePrintf("%s has declared a port (%d) reserved by Openshift", cut.String(), port.ContainerPort)
+				rogueContainers = append(rogueContainers, cut.String())
+				break
+			}
+		}
+
+		// Then verify that no container is listening to the reserved OCP ports
+		outStr, errStr, err := crclient.ExecCommandContainerNSEnter(getListeningPortsCmd, cut)
+		if err != nil || errStr != "" {
+			tnf.ClaimFilePrintf("Failed to execute command %s on %s, err: %s, errStr: %s", getListeningPortsCmd, cut, err, errStr)
+			failedContainers++
+			continue
+		}
+
+		listeningPorts := make(map[declaredandlistening.Key]bool)
+		declaredandlistening.ParseListening(outStr, listeningPorts)
```

Comment:
Maybe in a subsequent PR, we could make `ParseListening` a function that returns a `map[declaredandlistening.Key]bool`.  Just something to think about.

Reply:
Agree. I will refactor this in a new PR.